### PR TITLE
added liquibase, changed the repository implementation

### PR DESCRIPTION
**Pull request by NikitaLafinskiy**

### CODE REVIEW

**Review by ahoienko — CHANGES_REQUESTED** [review]
One of requirements is not implemented (Change spring.jpa.hibernate.ddl-auto property value to validate)

**Comment by NikitaLafinskiy** [comment]
> One of requirements is not implemented (Change spring.jpa.hibernate.ddl-auto property value to validate)

I did change it but I do not want to commit my application.properties file, it has a value of:
spring.jpa.hibernate.ddl-auto=validate

File: src/main/java/com/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +34,16 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
+
+    @PutMapping("{id}")
```
Comment (sarakhmen) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
    @PutMapping("/{id}")
```
fix in all places

File: src/main/java/com/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -32,4 +34,16 @@ public BookDto getBookById(@PathVariable Long id) {
...
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto createBookRequestDto) {
+        return bookService.updateById(id, createBookRequestDto);
+    }
+
+    @DeleteMapping("{id}")
+    public String deleteBookById(@PathVariable Long id) {
+        bookService.deleteById(id);
+        return "Deleted a book with an id of " + id;
```
Comment (sarakhmen) [inline src/main/java/com/bookstore/controller/BookController.java]: let's not return any strings from this method. Simply annotate it with  @ResponseStatus(value = HttpStatus.NO_CONTENT)

**Review by sarakhmen — CHANGES_REQUESTED** [review]
let's improve your solution :)

File: src/main/java/com/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +36,17 @@ public BookDto getBookById(@PathVariable Long id) {
...
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto createBookRequestDto) {
+        return bookService.updateById(id, createBookRequestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(value = HttpStatus.NO_CONTENT)
+    public String deleteBookById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
    public void deleteBookById(@PathVariable Long id) {
```

File: src/main/java/com/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -32,4 +36,17 @@ public BookDto getBookById(@PathVariable Long id) {
...
+        bookService.deleteById(id);
+        return "Deleted a book with an id of " + id;
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/controller/BookController.java]:
```suggestion
```

File: src/main/java/com/bookstore/entity/Book.java [hunk 5]
```diff
@@ -9,11 +9,15 @@
...
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Getter
 @Setter
 @Table(name = "books")
+@SQLDelete(sql = "UPDATE books SET is_deleted = false WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/entity/Book.java]: check if Where isn't deprecated

File: src/main/java/com/bookstore/service/BookServiceImpl.java [hunk 6]
```diff
@@ -37,4 +37,16 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toBookDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.toBookEntity(createBookRequestDto);
+        book.setId(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBookFromDto(createBookRequestDto, book);
```

File: src/main/resources/db/changelog/changes/01-book-init.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+              - column:
+                  name: description
+                  type: VARCHAR(255)
+              - column:
+                  name: cover_image
+                  type: VARCHAR(255)
+              - column:
+                  name: is_deleted
+                  type: BOOLEAN
+                  defaultValueBoolean: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-book-init.yaml]:
```suggestion
                  defaultValueBoolean: false
                  constraints:
                      nullable: false
```

File: src/main/resources/db/changelog/changes/01-book-init.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+  - changeSet:
+      id: 2
+      author: nikita
+      changes:
+        - addNotNullConstraint:
+            tableName: books
+            columnName: is_deleted
+            columnDataType: BOOLEAN
+            defaultValueBoolean: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-book-init.yaml]:
```suggestion
```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: classpath:db/changelog/changes/01-book-init.yaml
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: classpath:db/changelog/changes/01-create-books-table.yaml
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/bookstore/entity/Book.java [hunk 10]
```diff
@@ -35,4 +39,6 @@ public class Book {
 
     private String coverImage;
 
+    @Column(nullable = false)
+    private Boolean isDeleted;
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/entity/Book.java]:
```suggestion
    private boolean isDeleted;
```

File: src/main/java/com/bookstore/service/BookServiceImpl.java [hunk 11]
```diff
@@ -37,4 +37,15 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toBookDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.updateBookFromDto(createBookRequestDto, id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).orElseThrow...
        bookMapper.updateBookFromDto(createBookRequestDto, book);
```

File: src/main/java/com/bookstore/mapper/BookMapper.java [hunk 12]
```diff
@@ -5,10 +5,14 @@
...
+import org.mapstruct.Mapping;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
     BookDto toBookDto(Book book);
 
     Book toBookEntity(CreateBookRequestDto createBookRequestDto);
+
+    @Mapping(target = "id", source = "id")
+    Book updateBookFromDto(CreateBookRequestDto createBookRequestDto, Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/mapper/BookMapper.java]:
```suggestion
    void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstore/service/BookServiceImpl.java [hunk 13]
```diff
@@ -37,4 +37,18 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toBookDto)
                 .toList();
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(
+                "Unable to find a book with an id of "
+                        + id));
+        book = bookMapper.updateBookFromDto(createBookRequestDto, book);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/service/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBookFromDto(createBookRequestDto, book);
```

File: src/main/java/com/bookstore/mapper/BookMapper.java [hunk 14]
```diff
@@ -5,10 +5,13 @@
...
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
     BookDto toBookDto(Book book);
 
     Book toBookEntity(CreateBookRequestDto createBookRequestDto);
+
+    Book updateBookFromDto(CreateBookRequestDto createBookRequestDto, @MappingTarget Book book);
```
Comment (Elena-Bruyako) [inline src/main/java/com/bookstore/mapper/BookMapper.java]:
```suggestion
    void updateBookFromDto(CreateBookRequestDto createBookRequestDto, @MappingTarget Book book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**